python/lsst/cell_coadds/_uniform_grid.py

```diff
@@ -0,0 +1,217 @@
+        self._bbox = geom.Box2I(min, geom.Extent2I(cell_size.getX() * shape.x, cell_size.getY() * shape.y))
+
+    # Factory methods for constructing a UniformGrid
+    @classmethod
+    def from_bbox_shape(cls, bbox: geom.Box2I, shape: Index2D) -> UniformGrid:
+        """Factory method to construct from a bounding box and a shape.
+
+        Parameters
+        ----------
+        bbox : `lsst.geom.Box2I`
+            Bounding box of the full grid.
+        shape : `lsst.skymap.Index2D`
+            Number of cells in the grid in each dimension.
+            Must divide the ``bbox`` width and height evenly.
+
+        Returns
+        -------
+        grid : `UniformGrid`
+            A new UniformGrid instance.
+
+        Raises
+        ------
+        LengthError
+            Raised if ``shape`` dimensions do not divide the ``bbox``
+            dimensions evenly.
+        """
+        cls._validate_bbox_shape(bbox, shape)
+
+        cell_size = geom.Extent2I(bbox.getWidth() // shape.x, bbox.getHeight() // shape.y)
+        return cls(cell_size, shape, bbox.getMin())
+
+    @classmethod
+    def from_bbox_cell_size(cls, bbox: geom.Box2I, cell_size: geom.Extent2I) -> UniformGrid:
+        """Factor method to construct from a bounding box and a cell size.
+
+        Parameters
+        ----------
+        bbox : `lsst.geom.Box2I`
+            Bounding box of the full grid.
+        cell_size : `lsst.geom.Extent2I`
+            Size of each grid cell.
+            Must divide the ``bbox`` width and height evenly.
+
+        Returns
+        -------
+        grid : `UniformGrid`
+            A new UniformGrid instance.
+
+        Raises
+        ------
+        LengthError
+            Raised if ``cell_size`` dimensions do not divide the ``bbox``
+            dimensions evenly.
+        """
+        cls._validate_bbox_cell_size(bbox, cell_size)
+        shape = Index2D(bbox.getWidth() // cell_size.x, bbox.getHeight() // cell_size.y)
+        return cls(cell_size, shape, bbox.getMin())
+
+    # Methods to validate the input parameters
+    @staticmethod
+    def _validate_bbox_shape(bbox: geom.Box2I, shape: Index2D) -> None:
+        if bbox.getWidth() % shape.x != 0:
+            raise LengthError(
+                f"Bounding box width {bbox.getWidth()} is not evenly divided by x shape {shape.x}."
+            )
+        if bbox.getHeight() % shape.y != 0:
+            raise LengthError(
+                f"Bounding box height {bbox.getHeight()} is not evenly divided by y shape {shape.y}."
+            )
+
+    @staticmethod
+    def _validate_bbox_cell_size(bbox: geom.Box2I, cell_size: geom.Extent2I) -> None:
+        if bbox.getWidth() % cell_size.x != 0:
+            raise LengthError(
+                f"Bounding box width {bbox.getWidth()} is not evenly divided by x cell_size "
+                f"{cell_size.getX()}."
+            )
+
+        if bbox.getHeight() % cell_size.y != 0:
+            raise LengthError(
```

Comment:
This should probably just raise `ValueError`?

Reply:
Either that or `IndexError`.